spmimage/linear_model/mp.py

```diff
@@ -0,0 +1,59 @@
+"""Matching pursuit algorithms
+"""
+
+import numpy as np
+from sklearn.utils import check_array
+
+def matching_pursuit(dictionary, signal, n_nonzero_coefs=None, copy_dictionary=True, copy_signal=True, tol=None):
+    """Matching Pursuit (MP)
+    
+    Solves n_targets Matching Pursuit problems.
+    The goal is to find a sparse array `coefs` such that::
+        signal ~= coefs * dictionary
+    Parameters
+    ----------
+    dictionary : array, shape (n_components, n_features)
+        Input data. Columns are assumed to have unit norm.
+
+    signal : array, shape (n_samples, n_features)
+        Input targets.
+
+    n_nonzero_coefs : int
+        Targeted number of non-zero elements
+
+    copy_X : bool, optional
+        Whether the input data must be copied by the algorithm. A false
+        value is only helpful if it is already Fortran-ordered, otherwise a
+        copy is made anyway.
+
+    copy_y : bool, optional
+        Whether the covariance vector Xy must be copied by the algorithm.
+        If False, it may be overwritten.
+
+    tol : float
+        Maximum norm of the residual.
+    """
+    if tol is not None and tol < 0:
```

Comment:
@zalgo-edu Please make sure each vector in dictionary has norm 1.
If this condition is not satisfied, please raise warning message.